powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/controller/v3/UpgradeController.java

```diff
@@ -123,12 +124,13 @@ public Response upgradeCommit(@RequestHeader(value = PowerAuthSignatureHttpHeade
         try {
             PowerAuthSignatureHttpHeaderValidator.validate(header);
         } catch (InvalidPowerAuthHttpHeaderException ex) {
-            throw new PowerAuthUpgradeException(ex.getMessage());
+            logger.warn("Signature validation failed, error: {}", ex.getMessage());
+            throw new PowerAuthUpgradeException();
```

Comment:
Let's pass `ex` as a parameter to the new exception to keep the cause somewhere.